### PR TITLE
Do not show query / metadata options until dataset is saved

### PR DESCRIPTION
**Pull request by gusaiani**
tl;dr: This PR makes model-without-question-creation pages no longer display the `Query` and `Metadata` buttons in the top bar.

Please also look at https://github.com/metabase/metabase/pull/25877, another solution.

This is kind of a product question PR.

What should we do with the `Query` and `Metadata` buttons until a model is saved?

When a user decides to create a brand new model, they will land on a query page.

Now, what about metadata?
Until we have some firm query, a metadata makes little sense, correct?

Even if there is a coherent query, maybe the metadata will be changed before a save.
Does it make sense to even offer to edit metadata before a model is saved?
Especially considering that after a save we will offer the button to start editing metadata.

### How to Test

1. `+ New`
2. `Model`
3. `Native Query`

You won't see either the `Query` or the `Metadata` buttons in the top nav for models.

After saving the model, you will see both.

### CODE REVIEW

**Review by ranquild — COMMENTED** [review]
We could also "disable" the `Metadata` button until the query is run. That's the primary limitation we have right now, isn't it?

**Comment by gusaiani** [comment]
> We could also "disable" the Metadata button until the query is run. That's the primary limitation we have right now, isn't it?

Yes, that's also a possibility, and my first thought as well.
As I used the feature, it felt like metadata was working less often that it was working with queries in place etc, hence the possibility in this PR.

Happy to go the `disabled` way as well, it may indeed be better. Curious what @mazameli thinks.

**Comment by mazameli** [comment]
Metadata editing is a big part of the flow of creating a modal, so I'm 100% in favor of https://github.com/metabase/metabase/pull/25877 instead of this.

**Comment by gusaiani** [comment]
#25877 was preferred
